### PR TITLE
ninth task

### PR DESCRIPTION
**Pull request by YANGMAO-ZHANG**
@Felixnoo

### CODE REVIEW

**Comment by YANGMAO-ZHANG** [comment]
#1854

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 1]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
-| environment.description                    | string   | The description of environment variables.                    |
-| environment.optValues                      | []string | The list of parameters for environment variables.            |
-| environment.required                       | bool     | Whether the environment variable is required to be set.      |
-| environment.defaultValue                   | string   | The default value of environment variables.                  |
-| environment.value                          | string   | The value of environment variables.                          |
-| iconPath                                   | string   | The application name.                                        |
-| version                                    | string   | The version of S2I template.                                 |
-| description                                | string   | The description of the template's functions and usage.       |
+| *containerInfo                             | []struct | 关于镜像构建器的信息。                                       |
+| *containerInfo.builderImage                | string   | 镜像构建器，如：kubesphere/java-8-centos7:v2.1.0. S2I。      |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:273]: Delete `S2I。`

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 2]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
+| containerInfo.runtimeImage                 | string   | 运行时镜像，如：kubesphere/java-8-runtime:v2.1.0. S2I。      |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:274]: `运行时镜像` -> `S2I 运行时镜像`

Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:274]: Delete `S2I。`

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 3]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
+| containerInfo.buildVolumes                 | []string | 关于安装卷的信息。格式为 "volume_name:mount_path", 如："s2i_java_cache:/tmp/artifacts","test_cache:test_path"]。 |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:275]: `安装卷` -> `挂载卷`

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 4]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
+| containerInfo.runtimeArtifacts             | []struct | 输出制品的原始路径和目标路径；仅在分阶段构建中添加。         |
+| 制品facts.source                           | string   | 制品在镜像构建器的原始路径。                                 |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:277]: `制品facts.source` -> `containerInfo.runtimeArtifacts.source`

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 5]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
+| containerInfo.runtimeArtifacts.destination | string   | 运行时镜像中制品的目标路径。                                 |
+| containerInfo.runtimeArtifacts.keep        | bool     | 是否将数据保留在输出镜像中。                                 |
+| *defaultBaseImage                          | string   | 默认镜像构建器。                                             |
+| *codeFramework                             | string   | 代码框架类型，如：Java，Ruby。                               |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:281]: `，` -> `、`

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 6]
```diff
@@ -198,131 +203,91 @@ S2I scripts will use the flags defined in the Dockerfile as parameters. If you n
...
+| environment                                | []struct | 构建过程中的环境变量列表。                                   |
+| environment.key                            | string   | 环境变量的名称。                                             |
+| environment.type                           | string   | 环境变量键的类型。                                           |
+| environment.description                    | string   | 环境变量的描述。                                             |
+| environment.optValues                      | []string | 环境变量的参数列表。                                         |
+| environment.required                       | bool     | 是否需要设置环境变量。                                       |
+| environment.defaultValue                   | string   | 环境变量的默认值。                                           |
+| environment.value                          | string   | 环境变量的值。                                               |
+| iconPath                                   | string   | 应用名字。                                                   |
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:290]: `名字` -> `名称`

**Review by Felixnoo — CHANGES_REQUESTED** [review]
Please take a look at the comments, thanks.

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 7]
```diff
@@ -249,9 +249,13 @@ S2I 脚本将使用 Dockerfile 中定义的标志作为参数。如果您需要
 
    您可以在此位置访问 Nginx 应用程序：`http://localhost:8080`。
 
+<<<<<<< HEAD
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:252]: Please fix this conflict.

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 8]
```diff
@@ -280,11 +284,15 @@ spec:
  ```bash
    $ kubectl apply -f s2ibuildertemplate.yaml
    s2ibuildertemplate.devops.kubesphere.io/nginx created
-   ```
+ ```
 
+<<<<<<< HEAD
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:289]: Please fix this conflict.

**Review by Felixnoo — CHANGES_REQUESTED** [review]

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 9]
```diff
@@ -270,55 +267,55 @@ spec:
...
-   ```bash
+ ```bash
    $ kubectl apply -f s2ibuildertemplate.yaml
    s2ibuildertemplate.devops.kubesphere.io/nginx created
-   ```
-
-2. You can find the customized S2I template available in **Build Environment** when you create a S2I build on KubeSphere.
+ ```
+ 
+2. 在 KubeSphere 上创建 S2I 构建时，可以找到在**环境搭建**中找到自定义 S2I 模板。
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:284]: 可以找到在**环境搭建**
->
可以在**构建环境**

**Review by Felixnoo — CHANGES_REQUESTED** [review]
Hi, thanks for your effort. Just one more change.

File: content/zh/docs/project-user-guide/image-builder/s2i-templates.md [hunk 10]
```diff
@@ -9,7 +9,7 @@ weight: 10640
 
 当您了解了 Source-to-Image (S2I) 的工作流和逻辑，就可以根据您的项目自定义镜像构建器模板（即 S2I / B2I 模板），以扩展 S2I 功能。KubeSphere 提供了几种常见的镜像构建器模板，如 [Python ](https://github.com/kubesphere/s2i-python-container/)和  [Java](https://github.com/kubesphere/s2i-java-container/)。 
 
-本教程演示如何创建包含 Nginx 服务的镜像构建器。如果需要在项目中使用运行时镜像，请参阅[本文档](https://github.com/kubesphere/s2irun/blob/master/docs/runtime_image.md)以了解有关如何创建运行时映像的更多信息。
+本教程演示如何创建包含 Nginx 服务的镜像构建器。如果需要在项目中使用运行时镜像，请参阅[本文档](https://github.com/kubesphere/s2irun/blob/master/docs/runtime_image.md)以了解有关如何创建运行时镜像的更多信息。
```
Comment (Felixnoo) [inline content/zh/docs/project-user-guide/image-builder/s2i-templates.md:12]: Great catch here.

BTW, I see a lot of `映像` in this PR. I guess you used some CAT tool during translation, which is great, but you have to pay attention to the consistency of terms.

Comment (YANGMAO-ZHANG) [reply content/zh/docs/project-user-guide/image-builder/s2i-templates.md:12]: Thanks!

**Pull request merged**